versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/FetchValuesUsingOpenAddressing.java

```diff
@@ -163,4 +167,16 @@ Collection<Key> checkForKeys(
     }
     return keysForNextRound;
   }
+
+  private void checkSegmentIndexValidity(final int segment, final Key key) {
+    // We expect segmentForKey to wrap in case the current round number is large enough to overrun
+    // the segment list, so this shouldn't happen
+    if (keyListsArray.length <= segment) {
+      throw new IllegalStateException(
```

Comment:
Nit: `Preconditions.checkState` (provides string-formatting)

Reply:
Agreed, I should use the variadic `checkState` overload instead of this verbose construct.  However, after reflecting on your comment about the testability of this exception elsewhere in the review, I've removed this check (with more on the thought process in that comment's thread).